CMakeLists.txt

```diff
@@ -156,17 +167,51 @@ if(NGEN_WITH_NETCDF)
     add_compile_definitions(NETCDF_ACTIVE)
 endif()
 
+if(NGEN_WITH_EXTERN_CXX_MODELS)
+    # This is set because SLOTH will link to gtest by default,
+    # which causes a configure error.
+    set(PACKAGE_TESTS OFF)
+    add_external_subdirectory(
+        SOURCE "${NGEN_EXT_DIR}/sloth"
+        OUTPUT "${NGEN_EXT_DIR}/sloth/cmake_build"
```

Comment:
First thought, before I've read through the PR in full - if we're doing the build automatically, we want to just put the output under the binary directory, not the source tree